### ISSUE TITLE
onTap gesture doesn't work together with ripple

### ISSUE THREAD

**Issue by Jeanlo** [ReinBentdal/division#27]
When using ripple effect with onTap/onTapDown/onTapUp gesture, the function from onTap is never called on division 0.7.0. On changelog says it was fixed on version 0.6.4 for all gestures, and it works with every gesture but onTap ones.

Example:
```dart
Division(
	style: S()
	  ..background.color(Colors.red)
	  ..height(32)
	  ..ripple(true)
	  ..width(32),
	gesture: G()..onTap(() => print('Hello')),
	child: Container(),
),
```
It never prints Hello when ripple is true.

# flutter doctor
```
[√] Flutter (Channel stable, v1.7.8+hotfix.4, on Microsoft Windows [Version 10.0.18362.239], locale en-US)
    • Flutter version 1.7.8+hotfix.4 at C:\src\flutter
    • Framework revision 20e59316b8 (3 weeks ago), 2019-07-18 20:04:33 -0700
    • Engine revision fee001c93f
    • Dart version 2.4.0

 
[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at C:\Users\jeanl\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
    • All Android licenses accepted.

[√] Android Studio (version 3.4)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 38.1.1
    • Dart plugin version 183.6270
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[√] VS Code (version 1.36.1)
    • VS Code at C:\Users\jeanl\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.3.0

[√] Connected device (1 available)
    • SNE LX3 • DVR0218A08000797 • android-arm64 • Android 9 (API 28)

• No issues found
```

**Comment by john6847** [comment]
Hi, I'm having the same problem here.

**Comment by ReinBentdal** [comment]
Thanks for the detailed issue. I am currently working on a rewrite of Division. That is a higher priority for me at the moment. I will look into this issue when the rewrite is finished

**Comment by Jeanlo** [comment]
Thanks, keep the good work!

**Comment by ReinBentdal** [comment]
Thanks again for your help. The issue should be resolved in version 0.8.0
